spring-cloud-stream-schema/src/main/java/org/springframework/cloud/stream/schema/client/config/SchemaRegistryClientConfiguration.java

```diff
@@ -25,17 +29,28 @@
 
 /**
  * @author Marius Bogoevici
+ * @author Vinicius Carvalho
  */
 @Configuration
 @EnableConfigurationProperties(SchemaRegistryClientProperties.class)
+@EnableCaching
 public class SchemaRegistryClientConfiguration {
 
+	@Autowired
+	private SchemaRegistryClientProperties schemaRegistryClientProperties;
+
 	@Bean
+	@ConditionalOnMissingBean(SchemaRegistryClient.class)
 	public SchemaRegistryClient schemaRegistryClient(SchemaRegistryClientProperties schemaRegistryClientProperties) {
```

Comment:
`@ConditionalOnMissingBean` is unsafe for use outside an autoconfiguration in general. (We make an exception for it in `BindingServiceConfiguration` and that is exclusively for hierarchies).